src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/AzureWebAppDiagnosticsLoggerProvider.cs

```diff
@@ -13,45 +14,51 @@ public class AzureWebAppDiagnosticsLoggerProvider : ILoggerProvider
     {
         private readonly IWebAppLogConfigurationReader _configurationReader;
 
-        private readonly ILoggerProvider _innerLoggerProvider;
-        private readonly bool _runningInWebApp;
+        private readonly LoggerFactory _loggerFactory;
 
         /// <summary>
         /// Creates a new instance of the <see cref="AzureWebAppDiagnosticsLoggerProvider"/> class.
         /// </summary>
-        public AzureWebAppDiagnosticsLoggerProvider(WebAppContext context, int fileSizeLimitMb)
+        public AzureWebAppDiagnosticsLoggerProvider(WebAppContext context, AzureWebAppDiagnosticsSettings settings)
```

Comment:
Null check for the new arg. You'll get a null ref otherwise when you create the FileLoggerProvider